IllegalArgumentException if analysing file in root of Windows mountpoint

Given these Gradle files, if they are put in the root of a Windows mount point, the `subpath` extraction fails when running the `spotlessGroovyGradle` task.

`settings.gradle`:
```groovy
rootProject.name = 'bug' // required because of a Gradle bug/feature when the project is in the root
```

`build.gradle`:
```groovy
plugins {
    id 'com.diffplug.spotless' version '5.8.2'
}

spotless {
    groovyGradle {
        target('*.gradle')
        greclipse()
    }   
}
```

Relevant failure output:
```
$ gradle :spotlessGroovyGradle -s
> Task :spotlessGroovyGradle FAILED

Caused by: java.lang.IllegalArgumentException: Expected 'S:/build.gradle' to start with 'S://'
        at com.diffplug.spotless.FileSignature.subpath(FileSignature.java:203)
        at com.diffplug.spotless.extra.GitAttributesLineEndings$CachedEndings.endingFor(GitAttributesLineEndings.java:127)
        at com.diffplug.spotless.extra.GitAttributesLineEndings$RelocatablePolicy.getEndingFor(GitAttributesLineEndings.java:95)
        at com.diffplug.spotless.Formatter.computeLineEndings(Formatter.java:211)
        at com.diffplug.spotless.PaddedCell.calculateDirtyState(PaddedCell.java:203)
        at com.diffplug.spotless.PaddedCell.calculateDirtyState(PaddedCell.java:188)
        at com.diffplug.gradle.spotless.SpotlessTaskImpl.processInputFile(SpotlessTaskImpl.java:71)
        at com.diffplug.gradle.spotless.SpotlessTaskImpl.performAction(SpotlessTaskImpl.java:57)
```

Gradle version: 6.7.1
Spotless version: 5.8.2.
JDK version: 1.8.0_252
Operating System: Windows 10 in VirtualBox VM

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -23,6 +23,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 ### Fixed
 * ktfmt v0.19+ with dropbox-style works again ([#765](https://github.com/diffplug/spotless/pull/765)).
 * prettier no longer throws errors on empty files ([#751](https://github.com/diffplug/spotless/pull/751)).
+* fixed error when running on root of windows mountpoint ([#760](https://github.com/diffplug/spotless/pull/760)).
 
 ## [2.10.2] - 2020-11-16
 ### Fixed

--- a/lib/src/main/java/com/diffplug/spotless/FileSignature.java
+++ b/lib/src/main/java/com/diffplug/spotless/FileSignature.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2020 DiffPlug
+ * Copyright 2016-2021 DiffPlug
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -200,6 +200,9 @@ public final class FileSignature implements Serializable {
 		if (child.startsWith(root)) {
 			return child.substring(root.length());
 		} else {
+			if (machineIsWin() && root.endsWith("://") && child.startsWith(root.substring(0, root.length() - 1))) {
+				return child.substring(root.length() - 1);
+			}
 			throw new IllegalArgumentException("Expected '" + child + "' to start with '" + root + "'");
 		}
 	}

--- a/plugin-gradle/CHANGES.md
+++ b/plugin-gradle/CHANGES.md
@@ -14,6 +14,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 * `ratchetFrom` now works with git-submodule ([#746](https://github.com/diffplug/spotless/issues/746))
 * ktfmt v0.19+ with dropbox-style works again ([#765](https://github.com/diffplug/spotless/pull/765)).
 * prettier no longer throws errors on empty files ([#751](https://github.com/diffplug/spotless/pull/751)).
+* fixed error when running on root of windows mountpoint ([#760](https://github.com/diffplug/spotless/pull/760)).
 
 ## [5.8.2] - 2020-11-16
 ### Fixed

--- a/plugin-maven/CHANGES.md
+++ b/plugin-maven/CHANGES.md
@@ -16,6 +16,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 * Fix broken test for spotlessFiles parameter on windows ([#737](https://github.com/diffplug/spotless/pull/737))
 * ktfmt v0.19+ with dropbox-style works again ([#765](https://github.com/diffplug/spotless/pull/765)).
 * prettier no longer throws errors on empty files ([#751](https://github.com/diffplug/spotless/pull/751)).
+* fixed error when running on root of windows mountpoint ([#760](https://github.com/diffplug/spotless/pull/760)).
 
 ## [2.6.1] - 2020-11-16
 ### Fixed

--- a/testlib/src/test/java/com/diffplug/spotless/FileSignatureTest.java
+++ b/testlib/src/test/java/com/diffplug/spotless/FileSignatureTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2020 DiffPlug
+ * Copyright 2016-2021 DiffPlug
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -25,6 +25,7 @@ import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 
+import org.assertj.core.api.Assertions;
 import org.junit.Test;
 
 public class FileSignatureTest extends ResourceHarness {
@@ -78,5 +79,12 @@ public class FileSignatureTest extends ResourceHarness {
 	@Test
 	public void testSubpath() {
 		assertThat(FileSignature.subpath("root/", "root/child")).isEqualTo("child");
+	}
+
+	@Test
+	public void windowsRoot() {
+		org.junit.Assume.assumeTrue(FileSignature.machineIsWin());
+		String subpath = FileSignature.subpath("S://", "S:/build.gradle");
+		Assertions.assertThat(subpath).isEqualTo("build.gradle");
 	}
 }

